### PR TITLE
call boogie in a safer way, at the temporary cost of disabling counterexamples

### PR DESCRIPTION
**Pull request by typerSniper**

### CODE REVIEW

**Comment by typerSniper** [comment]
@atomb

File: Sources/SymDiff/source/BoogieVerify.cs [hunk 1]
```diff
@@ -909,28 +910,47 @@ public static int RunVerificationTask(VerificationTask vt, Program prog, out boo
...
-                if (BoogieUtils.ResolveAndTypeCheckThrow(prog, Options.MergedProgramOutputFile, BoogieUtils.BoogieOptions))
-                    return 1;
-
-                newEq = vt.Eq;
-                newProg = prog;
+                prog = BoogieUtils.ParseProgram(rs_filename);
+                var stringWriter = new StringWriter();
+                var success = engine.ProcessProgram(stringWriter, prog, rs_filename).Result;
+                string output = stringWriter.ToString();
+                var verified = output.Contains("0 errors");
```
Comment (atomb) [inline Sources/SymDiff/source/BoogieVerify.cs]: I think this should be `" 0 errors"`, with a space at the beginning, or we might consider `20 errors` as success.

File: Sources/SymDiff/source/BoogieVerify.cs [hunk 2]
```diff
@@ -909,28 +910,47 @@ public static int RunVerificationTask(VerificationTask vt, Program prog, out boo
...
                 {
-                    Log.Out(Log.Verifier, "Parse Error!!! in   " + vt.Eq.Name);
-                    return 1;
+                    RunningBoogieFromCommandLine = true
+                };
+                options.Parse(["/soundLoopUnrolling", "/inline:assume"]);
+                var resultCache = new VerificationResultCache();
+                var engine = new ExecutionEngine(options, resultCache);
+                if (engine == null) {
+                    Log.Out (Log.Verifier, "BIG ERRROR ENGINE");
```
Comment (atomb) [inline Sources/SymDiff/source/BoogieVerify.cs]:
```suggestion
                    Log.Out (Log.Verifier, "Failed to create execution engine!");
```

File: Sources/SymDiff/source/BoogieVerify.cs [hunk 3]
```diff
@@ -909,28 +910,47 @@ public static int RunVerificationTask(VerificationTask vt, Program prog, out boo
...
+                }
+                Log.Out(Log.Verifier, "DONE");
+                // newEq = vt.Eq;
+                // newProg = prog;
 
-                var vcgen = InitializeVC(newProg);
-                //SDiff.Boogie.Process.ResolveAndTypeCheck(newProg, "");
-                newDict = SDiff.Boogie.Process.BuildProgramDictionary(newProg.TopLevelDeclarations.ToList());
+                // var vcgen = InitializeVC(newProg);
+                // //SDiff.Boogie.Process.ResolveAndTypeCheck(newProg, "");
```
Comment (atomb) [inline Sources/SymDiff/source/BoogieVerify.cs:959]: Here and below the additional comment characters are unnecessary.

**Review by atomb — CHANGES_REQUESTED** [review]

File: Sources/SymDiff/source/BoogieVerify.cs (hunk 3 above)
Comment (typerSniper) [reply Sources/SymDiff/source/BoogieVerify.cs:959]: I feel they are temporarily useful, so i am going to leave them in. My goal is to uncomment these things for cexs.

**Review by atomb — APPROVED** [review]
This looks good. Just noting, for the record, that we'll have a future PR to re-enable filling the `vt.Counterexamples` field.

**Pull request merged**